packages/vaadin-combo-box/src/vaadin-combo-box-dropdown-wrapper.js

```diff
@@ -195,37 +199,11 @@ class ComboBoxDropdownWrapperElement extends PolymerElement {
 
   _getItems(opened, items) {
     if (opened) {
-      if (this._isNotEmpty(items) && this._selector && !this.filterChanged) {
-        // iron-list triggers the scroller's reset after items update, and
-        // this is not appropriate for undefined size lazy loading.
-        // see https://github.com/vaadin/vaadin-combo-box-flow/issues/386
-        // We store iron-list scrolling position in order to restore
-        // it later on after the items have been updated.
-        const currentScrollerPosition = this._selector.firstVisibleIndex;
-        if (currentScrollerPosition !== 0) {
-          this._oldScrollerPosition = currentScrollerPosition;
-          this._resetScrolling = true;
-        }
-      }
-      // Let the position to be restored in the future calls unless it's not
-      // caused by filtering
-      this.filterChanged = false;
       return items;
     }
     return [];
   }
 
-  _restoreScrollerPosition(items) {
-    if (this._isNotEmpty(items) && this._selector && this._oldScrollerPosition !== 0) {
-      // new items size might be less than old scrolling position
-      this._scrollIntoView(Math.min(items.length - 1, this._oldScrollerPosition));
-      this._resetScrolling = false;
-      // reset position to 0 again in order to properly handle the filter
-      // cases (scroll to 0 after typing the filter)
-      this._oldScrollerPosition = 0;
-    }
-  }
-
```

Comment:
All logic related to restoring the scroll position on items change can be removed from combo-box, because the virtualizer has the same functionality built-in.